### ISSUE TITLE
Style/MethodName error on camel-cased class method

### ISSUE THREAD

**Issue by matthewd** [bbatsov/rubocop#2285]
```
0.34.2 (using Parser 2.2.2.6, running on ruby 2.2.2 x86_64-darwin14)
```

Style/MethodName doesn't seem to like:

```ruby
class X
  def self.fooBar
  end
end
```

**Comment by bbatsov** [comment]
That's the expected behaviour. I'm not sure why do you think this is a problem?

**Comment by matthewd** [comment]
Sorry, this was indeed unclear. With autocorrect enabled, it *errors*:

```
An error occurred while Style/MethodName cop was inspecting ../x.rb.
undefined method `class_type?' for nil:NilClass
```

**Comment by bbatsov** [comment]
Hmm, I can't reproduce this.

**Comment by matthewd** [comment]
Weird. It doesn't seem environment-specific for me... I'm not using any local config, and I just installed rubocop on a random box:

```
bravo:~% ruby -v
ruby 2.1.2p95 (2014-05-08) [x86_64-linux-gnu]
bravo:~% sudo gem install rubocop
[..]
Successfully installed rubocop-0.34.2
bravo:~% > x.rb <<EOF     
class X
  def self.fooBar
  end
end
EOF
bravo:~% rubocop --only MethodName -a ./x.rb
warning: parser/current is loading parser/ruby21, which recognizes
warning: 2.1.7-compliant syntax, but you are running 2.1.2.
Inspecting 1 file
An error occurred while Style/MethodName cop was inspecting /home/matthew/x.rb.
To see the complete backtrace run rubocop -d.
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/MethodName cop was inspecting /home/matthew/x.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.34.2 (using Parser 2.3.0.pre.2, running on ruby 2.1.2 x86_64-linux-gnu)
bravo:~%
```
